tests/blazesym.rs

```diff
@@ -638,30 +693,56 @@ fn normalize_build_id_rading() {
 /// Check that we can look up an address.
 #[test]
 fn inspect() {
-    fn test(src: inspect::Source) {
+    fn test(src: inspect::Source, no_vars: bool) {
         let inspector = Inspector::new();
         let results = inspector
-            .lookup(&src, &["factorial"])
-            .unwrap()
-            .into_iter()
-            .flatten()
-            .collect::<Vec<_>>();
-        assert_eq!(results.len(), 1);
+            .lookup(&src, &["factorial", "a_variable"])
+            .unwrap();
+        assert_eq!(results.len(), 2);
 
         let result = &results[0];
-        assert_eq!(result.addr, 0x2000100);
-        assert_ne!(result.file_offset, None);
+        assert_eq!(result.len(), 1);
+        assert_eq!(result[0].addr, 0x2000100);
+        assert_eq!(result[0].sym_type, SymType::Function);
+        assert_ne!(result[0].file_offset, None);
         assert_eq!(
-            result.obj_file_name.as_deref().unwrap(),
+            result[0].obj_file_name.as_deref().unwrap(),
             src.path().unwrap()
         );
+
+        let result = &results[1];
+        if no_vars {
+            assert!(result.is_empty(), "{result:#x?}");
+        } else {
+            assert_eq!(result.len(), 1);
+            assert_eq!(result[0].addr, 0x2001100);
+            assert_eq!(result[0].sym_type, SymType::Variable);
+            assert_ne!(result[0].file_offset, None);
+            assert_eq!(
+                result[0].obj_file_name.as_deref().unwrap(),
+                src.path().unwrap()
+            );
+        }
     }
 
     let test_dwarf = Path::new(&env!("CARGO_MANIFEST_DIR"))
         .join("data")
         .join("test-stable-addresses-dwarf-only.bin");
     let src = inspect::Source::Elf(inspect::Elf::new(test_dwarf));
-    let () = test(src);
+    // Our `DwarfResolver` type does not currently support look up of
+    // variables.
+    let no_vars = true;
+    let () = test(src, no_vars);
+
+    let test_elf = Path::new(&env!("CARGO_MANIFEST_DIR"))
+        .join("data")
+        .join("test-stable-addresses.bin");
```

Comment:
this is so common, that I feel like it would be justified to add a macro or function that would construct this from normal path specifier. I.e.,

```
let tst_elf = test_path("data/test-stable-addresses.bin")
```

(and function would take care of / vs \ on windows, and joining, etc.

Reply:
Yeah, we can come up with something. Let me do that as a follow up.